plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,319 @@
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, tableName, tableLocation));
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + " (a, b, c) COMMENT 'my-table-comment' AS VALUES (1, 'INDIA', true)");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the table content to register a table
+        dropTableFromMetastore(tableName);
+
+        assertQuerySucceeds("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertThat(getTableComment(tableName)).isEqualTo("my-table-comment");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithDifferentTableName()
+    {
+        String tableName = "test_register_table_with_different_table_name_old_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + " AS SELECT 1 as a, 'INDIA' as b, true as c");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the table content to register a table
+        dropTableFromMetastore(tableName);
+
+        String tableNameNew = "test_register_table_with_different_table_name_new_" + randomNameSuffix();
+        assertQuerySucceeds(format("CALL %s.system.register_table('%s', '%s', '%s')", CATALOG_NAME, SCHEMA, tableNameNew, tableLocation));
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableNameNew);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew.replaceFirst(tableNameNew, tableName));
+        assertQuery("SELECT * FROM " + tableNameNew, "VALUES (1, 'INDIA', true)");
+
+        assertUpdate(format("DROP TABLE %s", tableNameNew));
+    }
+
+    @Test
+    public void testRegisterEmptyTable()
+    {
+        String tableName = "test_register_table_with_no_data_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + "(a INT, b VARCHAR, c BOOLEAN)");
+
+        String tableLocation = getTableLocation(tableName);
+        String tableNameNew = tableName + "_new";
+
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, tableNameNew, tableLocation));
```

Comment:
shouldn't we drop before the table from the metastore?
Having two tables registered to the same location is not something we should encourage